core/src/main/scala/kafka/coordinator/transaction/DelayedTxnMetadataUpdate.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.coordinator.transaction
+
+import kafka.server.DelayedOperation
+import org.apache.kafka.common.protocol.Errors
+
+/**
+  * Delayed transaction state change operations that are added to the purgatory without timeout (i.e. these operations should never time out)
```

Comment:
In general, I wonder if it is a good idea to have an indefinite purgatory. Seems like it's safer to have timeouts and retry if we need to. For example, we wouldn't want user requests parked in the purgatory forever. It would be better to send a timeout error and let the client try again.